extensions/unicode-symbols/src/use-recently-used-items.ts

```diff
@@ -0,0 +1,85 @@
+import { getLocalStorageItem, setLocalStorageItem } from "@raycast/api";
+import { Dispatch, SetStateAction, useCallback, useEffect, useState } from "react";
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const useStateAndLocalStorage = <T, _ = void>(
+  key: string,
+  initialValue: T
+): [T, Dispatch<SetStateAction<T>>, boolean] => {
+  const [state, setState] = useState<T>(initialValue);
+  const [ready, setReady] = useState(false);
+
+  useEffect(() => {
+    // FIXME In the future version, we don't need didUnmount checking
+    // https://github.com/facebook/react/pull/22114
+    let didUnmount = false;
+
+    (async () => {
+      const cache = await getLocalStorageItem(key);
+
+      if (typeof cache === "string") {
+        if (!didUnmount) {
+          setState(JSON.parse(cache));
+          setReady(true);
+        }
+      } else {
+        if (!didUnmount) {
+          setReady(true);
+        }
+      }
+    })();
+
+    return () => {
+      didUnmount = true;
+    };
+  }, []);
+
+  const setStateAndLocalStorage = useCallback((updater) => {
+    setState((state) => {
+      const newValue = typeof updater === "function" ? updater(state) : updater;
+      setLocalStorageItem(key, JSON.stringify(newValue));
+      return newValue;
+    });
+  }, []);
+
+  return [state, setStateAndLocalStorage, ready];
+};
+
+interface UseRecentlyUsedParams<T> {
+  key: string;
+  comparator: keyof T | ((existingItem: T, itemToAdd: T) => boolean);
+  limit?: number;
+}
+
+export function useRecentlyUsedItems<T>(params: UseRecentlyUsedParams<T>) {
```

Comment:
Eventually, this can be made generic and moved to a commonly shared util in the future. I noticed other extensions are using a similar approach 👍